plugin.video.mytv_bg/addon.py

```diff
@@ -1,323 +1,407 @@
-            dialog.ok(__lang__(30003), e.code)
-            return
-        
-        data_result = response.read()
-        
-        try:
-            res = json.loads(data_result)
-        except Exception, e:
-            xbmc.log('%s addon: %s' % (__addon_name__, e))
-            return
-        
-        if 'token' in res:
-            return res['token']
-        
-        if 'status' in res:
-            if res['login'] == 'yes':
-
-                if self.login_iteration > 0:
-                    self.login_iteration = 0
-                    return
-                
-                self.logIN()
-                self.data=self.getLive()
-                data_new=self.getData(url)
-                self.login_iteration += 1
-                
-                return data_new
-
-            else:
-
-                dialog = xbmcgui.Dialog()
-                dialog.ok(__lang__(30003), res['msg'])
-                return
-        
-        return res
-
-    def getLive(self):
-        return {'token': self.token}
-
-    def makeUserPass(self):
-        return {
-            "usr":self.usr,
-            "pwd":self.pas,
-            "access_type":"xbmc_kodi",
-            "device_info": json.dumps({
-                "board":xbmc.getInfoLabel("System.BuildVersion"),
-                "brand":"xbmc/kodi " + xbmc.getInfoLabel("System.ProfileName"),
-                "device":xbmc.getInfoLabel("System.KernelVersion"),
-                "display":xbmc.getInfoLabel("System.FriendlyName"),
-                "model":xbmc.getInfoLabel("System.BuildDate"),
-                "product":"",
-                "push_id":"",
-                "uuid":""
-            })
-        }
-
-    def writeInFile(self):
-        fopen = open(__token_filepath__, "w+")
-        fopen.write(self.token + " " + plugin.get_setting('username'))
-        fopen.close()
-
-    def openReadFile(self):
-        if os.path.isfile(__token_filepath__):
-            fopen = open(__token_filepath__, "r")
-            temp_token = fopen.read()
-            fopen.close()
-            if temp_token:
-				arr = temp_token.partition(" ")
-				self.token = arr[0]
-				if arr[1] and arr[1] != plugin.get_setting('username'):
-					self.token = '';
-				temp_token = ''
-        else:
-            self.writeInFile()
-        
-        
-    def __log(self, text):
-        xbmc.log('%s addon: %s' % (__addon_name__, text))
-
-# END # LOGIN
-
-
-# LOG Method
-def __log(text):
-    xbmc.log('%s addon: %s' % (__addon_name__, text))
-    
-
-if __name__ == '__main__':
-    plugin.run()
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+#imports
+from xbmcswift import Plugin, xbmc, xbmcaddon, xbmcplugin, xbmcgui, clean_dict
+from urllib2 import urlopen, HTTPError
+import urlparse
+import xbmcgui
+from pyxbmct.addonwindow import *
+import sys
+import json
+import os.path
+from urllib2 import Request
+import urllib
+import urllib2
+import datetime
+import time
+
+__addon__ 			= xbmcaddon.Addon()

```

Comment:
This spacing is something you can improve in the future as well to improve readability